src/conversion.rs

```diff
@@ -34,156 +37,286 @@ impl fmt::Display for Error {
-            self.0.iter().map(|c| create_geo_coordinate(&c)).collect();
+    fn from(line_string: LineString<T>) -> Self {
+        let coords = line_string
+            .0
+            .into_iter()
+            .map(geo_types::Coordinate::from)
+            .collect();
+
+        geo_types::LineString(coords)
+    }
+}
 
-        geo_linestring.into()
+impl<'a, T> From<&'a MultiLineString<T>> for geo_types::Geometry<T>
+where
+    T: num_traits::Float,
+{
+    fn from(multi_line_string: &'a MultiLineString<T>) -> geo_types::Geometry<T> {
+        Self::MultiLineString(multi_line_string.clone().into())
     }
 }
 
-impl<'a, T> Into<geo_types::Geometry<T>> for &'a MultiLineString<T>
+impl<T> From<MultiLineString<T>> for geo_types::MultiLineString<T>
 where
     T: num_traits::Float,
 {
-    fn into(self) -> geo_types::Geometry<T> {
-        let geo_multilinestring: geo_types::MultiLineString<T> = self
+    fn from(multi_line_string: MultiLineString<T>) -> geo_types::MultiLineString<T> {
+        let geo_line_strings: Vec<geo_types::LineString<T>> = multi_line_string
             .0
-            .iter()
-            .map(|l| {
-                l.0.iter()
-                    .map(|c| create_geo_coordinate(&c))
-                    .collect::<Vec<_>>()
-            })
+            .into_iter()
+            .map(geo_types::LineString::from)
             .collect();
 
-        geo_multilinestring.into()
+        geo_types::MultiLineString(geo_line_strings)
     }
 }
 
-fn w_polygon_to_g_polygon<T>(polygon: &Polygon<T>) -> geo_types::Polygon<T>
+impl<'a, T> From<&'a Polygon<T>> for geo_types::Geometry<T>
 where
     T: num_traits::Float,
 {
-    let mut iter = polygon.0.iter().map(|l| {
-        l.0.iter()
-            .map(|c| create_geo_coordinate(c))
-            .collect::<Vec<_>>()
-            .into()
-    });
+    fn from(polygon: &'a Polygon<T>) -> geo_types::Geometry<T> {
+        Self::Polygon(polygon.clone().into())
+    }
+}
 
-    match iter.next() {
-        Some(interior) => geo_types::Polygon::new(interior, iter.collect()),
-        None => geo_types::Polygon::new(geo_types::LineString(vec![]), vec![]),
+impl<T> From<Polygon<T>> for geo_types::Polygon<T>
+where
+    T: num_traits::Float,
+{
+    fn from(polygon: Polygon<T>) -> Self {
+        let mut iter = polygon.0.into_iter().map(geo_types::LineString::from);
+        match iter.next() {
+            Some(interior) => geo_types::Polygon::new(interior, iter.collect()),
+            None => geo_types::Polygon::new(geo_types::LineString(vec![]), vec![]),
+        }
     }
 }
 
-impl<'a, T> Into<geo_types::Geometry<T>> for &'a Polygon<T>
+impl<'a, T> TryFrom<&'a MultiPoint<T>> for geo_types::Geometry<T>
 where
     T: num_traits::Float,
 {
-    fn into(self) -> geo_types::Geometry<T> {
-        w_polygon_to_g_polygon(self).into()
+    type Error = Error;
+
+    fn try_from(multi_point: &'a MultiPoint<T>) -> Result<Self, Self::Error> {
```

Comment:
I've inverted the existing `Into` implements to be `From` which provides `A::from(b)` while implicitly generating a `b.into(a)` to maintain existing functionality.

In general I'm not a fan of the implicit cloning that happens with this kind of type signature, which returns an owned geometry from a refernce to a wkt, but I wasn't feeling bold enough to break the existing API.